reader.go

```diff
@@ -1388,6 +1388,32 @@ func (r *Reader) SetOffset(offset int64) error {
 	return err
 }
 
+// SetOffsetWithTimestamp changes the offset from which the next batch of messages will be
+// read given the timestamp t.
+//
+// The method fails if the unable to connect to the coordinator, or unable to read the offset
+// given the ts, or if the reader has been closed.
+func (r *Reader) SetOffsetWithTimestamp(t time.Time) error {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	if r.closed {
+		return io.ErrClosedPipe
+	}
+
+	conn, err := r.coordinator()
```

Comment:
I'm confused about why we're looking for the coordinator here. The coordinator is supposed to be used when the reader is part of a consumer group, but after we're callign `SetOffset` which is a method that cannot be used in a consumer group.

Can you describe a bit more what your intent was here?